fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java

```diff
@@ -301,20 +309,25 @@ public static Range<PartitionKey> transferRange(Range<PartitionKey> baseRange,
             truncUpperDateTime = upperDate.toLocalDateTime();
         } else {
             upperDate = (DateLiteral) upperExpr;
-            truncUpperDateTime = getLowerDateTime(upperDate.toLocalDateTime(), granularity);
+            truncUpperDateTime = getUpperDateTime(upperDate.toLocalDateTime(), granularity);
```

Comment:
why ?

Reply:
explain in the PR comments: use getUpperDateTime rather than getLowerDateTime, since it generates a closedOpen range:

```
         result.put(mvPartitionName, Range.closedOpen(lowerPartitionKey, upperPartitionKey));
```